content/blog/2018/02/2018-02-22-cheetah.adoc

```diff
@@ -0,0 +1,134 @@
+:tags:
+- pipeline
+- performance
+- scalability
+- features
+:author: svanoort
+---
+
+:toc: 
+
+== The Problem To Solve
+Since it launched, Pipeline has had a bit of a Dr. Jekyll and Mr. Hyde performance problem.  In certain circumstances, Pipeline turns from a mild-mannered CI/CD assistant into a monster.  It will happily eat storage read/write capacity like popcorn with nary a care for the other concerns of our friendly butler.  When combined with additional factors, this results in real-world stability problems.  For example, combining slow storage with a spike in running Pipelines has brought down production Jenkins at more than one company.  Similarly users see issues if a busy master gets hit with an extra source of stress; past culprits have been heavy (ab)use of automated APIs, now-solved user lookup bugs, backup jobs, and plugins run amuck that load excessive numbers of builds.
+
+Now I'm _not_ saying this to scare people or to criticize the technology I've helped build. Implementing link:/blog/2017/02/01/pipeline-scalability-best-practice/[Pipeline scalability best practices] coupled with SSD storage keeps Jenkins in a happy place.  We just need context on the weaknesses to see why it's important to address them.
+
+== Introducing Project Cheetah
+
+Today we're announcing the first major results of Project Cheetah, our long-running effort to address these challenges and improve Pipeline scalability.  More broadly, Cheetah aims to help in 3 places:
+
+* *Small-scale containers:* Pipeline needs to run leanly in resource-constrained containers, to enable easy scale-out without consuming excessive resources on shared container hosts.
+* *Enterprise systems:* Pipeline needs to effectively serve high-scale Jenkins instances that are central to many large companies.
+* *General case:* run Pipelines a bit more quickly on average, and allow users to get much-stronger performance in worst-case scenarios.
+
+== Yes, but what does it DO? 
+
+Project Cheetah offers several things, but the most important is Durability Settings for all Pipelines, and especially the Performance-Optimized setting.  This setting avoids several unexpected performance "surprises" that may strike users.  In the general case, it greatly reduces the disk IO needs for Pipeline.  How much?  Below is a graph of storage utilization with legacy Pipeline versions (think early 2017) and with the latest version using the Performance-Optimized mode.  These are tested on an AWS instance backed by an EBS volume provisioned with 300 IOPs. 
+
+Before and After:
+
+image:/images/post-images/2018-02-22-cheetah/io-utilization.png[role="center"]
+
+As you can see, storage utilization goes down by a _lot_.  While the exact number will vary, across the benchmark testcases *this translates to 2x to 6x the Pipeline throughput before becoming IO-bound*. This also increases stability of Jenkins masters because they will tolerate unexpected load.
```

Comment:
> this translates to 2x to 6x the Pipeline throughput

this translates to __from__ 2x to 6x __of__ the Pipeline throughput ?

Reply:
good point, rephrasing this a bit